providers/FeedProvider.tsx

```diff
@@ -1,42 +1,53 @@
 'use client';
-import useFeed from '@/hooks/useFeed';
-import { createContext, useContext, useMemo, useState } from 'react';
+import { FeedFilter } from '@/types/Feed';
+import { Cast as CastType } from '@/types/Cast';
+import { ReactNode, createContext, useContext, useState } from 'react';
 
-const FeedContext = createContext<any>(null);
+type FeedProviderType = {
+  copied: boolean;
+  setCopied: (copied: boolean) => void;
+  fundsRecipient: `0x${string}` | undefined;
+  setFundsRecipient: (fundsRecipient: `0x${string}`) => void;
+  saleStrategy: `0x${string}` | undefined;
+  setSaleStrategy: (saleStrategy: `0x${string}`) => void;
+  setupActions: string[];
+  setSetupActions: (actions: string[]) => void;
+  filter: FeedFilter;
+  updateFilter: (change: FeedFilter) => void;
+  feed: CastType[];
+  setFeed: (feed: CastType[]) => void;
+};
```

Comment:
what are all of these?
- copied
- fundsRecipient
- saleStrategy
- setupActions